Selects & Ghost Fields

ludwiczakpawel opened the pull request:
## What type of PR is this? (check all applicable)

- [x] Feature

## Description

This PR does two things:

1. It brings "ghost" styling for form textfields. It basically gets rid of ANY decorations and visuals - it's just an editable box. 
2. It brings styling for `<select>` elements.

Both will be utilized for "write a post" new design (in fact, it's already used but I'm trying to make my other PR smaller :D).

## Screenshots

![image](https://user-images.githubusercontent.com/108287/80801676-6a76e000-8bad-11ea-9c1a-5342bc0b08d4.png)

## Related Tickets & Documents

- #7524 

## Added tests?

- [x] no, because they aren't needed

## Added to documentation?

- [x] no documentation needed

rhymes commented:
Can you provide a screenshot :D ? As I'm not sure how to visualize this :D

Ridhwana commented:
Hmm same here: it looks like there should have been a classname `crayons-select` to test with on the user misc page but it's not showing up on the element so its difficult to see the effects of the css.

```
    <div class="crayons-field">
      <%= f.label :inbox_type, "Inbox type", class: "crayons-field__label" %>
      <p class="crayons-field__description">Open your inbox to messages from people you don't follow or keep your inbox private to mutual follows.</p>
      <%= f.select :inbox_type, [%w[Open open], %w[Private private]], class: "crayons-select" %>
    </div>
```

<img width="1316" alt="Screenshot 2020-04-30 at 11 23 45" src="https://user-images.githubusercontent.com/2786819/80695125-bce4cd80-8ad5-11ea-8829-0b01dac151be.png">

Ridhwana commented on app/assets/stylesheets/components/forms.scss:
```diff
@@ -183,6 +219,18 @@
 }
 
+///////////////////////////////////////////////////
+.crayons-textfield {
```
Is a text field an input field or a text area?

ludwiczakpawel replied:
Correct, you can use `crayons-textfield` for both: `<input type="text">` and `<textarea>`

ludwiczakpawel commented:
@rhymes @Ridhwana I've added screenshot to the PR description.

>  it looks like there should have been a classname crayons-select to test with on the user misc page but it's not showing up on the element so it's difficult to see the effects of the css.

I believe there was some syntax error. Even though it wasn't part of this PR, I've fixed it here since it's a quick fix.

Ridhwana commented on app/views/pages/crayons.html.erb:
```diff
@@ -277,22 +277,54 @@
   <pre><%= '
   <textarea class="crayons-textfield" placeholder="This is palceholder..."></textarea>
   <textarea class="crayons-textfield" placeholder="This is palceholder..." disabled>Disabled textarea</textarea>
+  <textarea class="crayons-textfield crayons-textfield--ghost" placeholder="This is palceholder for ghost textarea..."></textarea>
```
```suggestion
  <textarea class="crayons-textfield crayons-textfield--ghost" placeholder="This is placeholder for ghost textarea..."></textarea>
```

small typo 😉

ludwiczakpawel replied:
thanks, fixed!

Ridhwana commented on app/views/pages/crayons.html.erb:
```diff
@@ -277,22 +277,54 @@
 
   <pre><%= '
   <textarea class="crayons-textfield" placeholder="This is palceholder..."></textarea>
   <textarea class="crayons-textfield" placeholder="This is palceholder..." disabled>Disabled textarea</textarea>
```
LOL copy and paste typos in this file on the placeholder key ```placeholder="This is palceholder...">```

ludwiczakpawel replied:
:)
